library/core/src/ptr/const_ptr.rs

```diff
@@ -1034,12 +1166,43 @@ impl<T: ?Sized> *const T {
         }
 
         // SAFETY:
-        // It is permisseble for `align_offset` to always return `usize::MAX`,
+        // It is permissible for `align_offset` to always return `usize::MAX`,
         // algorithm correctness can not depend on `align_offset` returning non-max values.
         //
         // As such the behaviour can't change after replacing `align_offset` with `usize::MAX`, only performance can.
         unsafe { intrinsics::const_eval_select((self, align), ctfe_impl, rt_impl) }
     }
+
+    /// Returns whether the pointer is properly aligned for `T`.
+    #[must_use]
+    #[inline]
+    #[unstable(feature = "pointer_is_aligned", issue = "none")]
+    pub fn is_aligned(self) -> bool
+    where
+        T: Sized,
+    {
+        self.addr() % core::mem::align_of::<T>() == 0
```

Comment:
Can be just `self.is_aligned_to(core::mem::align_of::<T>())`. I think it's easier to reason about.

Reply:
At first I thought that too, but actually no. At least I don't think we should (because of the panic that is not needed here)

Reply:
There is no way in the world the panic won't be optimized out (https://rust.godbolt.org/z/eTsc64TPG) - the alignment is a constant and those functions are small.